### ISSUE TITLE
Four (probably very easy) questions how how to use dgrid and on best practices

### ISSUE THREAD

**Issue by mercmobily** [SitePen/dgrid#485]
Hi, 

I am diving into dgrid (after too much server-side Node programming). 
I wrote something very basic, but I have questions... some of which might be really, really, really straightforward. I know there are 524234 ways of doing things... I am just after the best practices. I have seen too much bad code in my life, and I don't want to contribute to the disaster. 

Imagine that the "options" store returns a bunch of records, and I want to trigger an editing window when you click on them; so: 

```
var Contacts = declare('hotplate/bd/Contacts', [ _WidgetBase, _Container, _ContentPaneResizeMixin, _TemplatedMixin, _WidgetsInTemplateMixin,  ], { 

    templateString: jade.compile('' + 
    'div\n'+ 
    '  div(data-dojo-attach-point="dijit/layout/ContentPane" )\n' + 
    '    div(data-dojo-attach-point="list")\n' + 
    '')(),  

    postCreate: function(){ 
      this.inherited(arguments); 

      var list = new OnDemandList( { 
        store: stores('options', { workspaceIdCall: globals.workspaceId } ), 
        renderRow: function(item, options){ 
          return domConstruct.create('div', { innerHTML:JSON.stringify(item) } ); 
        }, 

      }, this.list); 

      list.on('.dgrid-row:click', function( e ) { 
        console.log("Clicked!"); 
        console.log( e ); 
        console.log( this ); 
        console.log( list.row( this ) ); 
      }); 

      list.startup(); 
    }, 

  }); 

 return Contacts; 
}); 
```

1) Can you see anything horribly wrong with what I did? (Although I didn't do too much) 

2) (IGNORE THIS ONE, I just realised that it's _just_ Dojo/on)  When I write `list.on('.dgrid-row:click', function( e ) {`, does the query search for elements of class .dgrid-row _within_ list.domNode? Or does it search the DOM tree?  

2a) (IGNORE THIS ONE, Ijust noticed that dojo/query can be given a context) Question 2 triggered this one: in Dojo we have, and use, dojo/query. Is it possible, in using dojo/query, to narrow down the search to a specific domNode? 

3) At click time, I will create an editing widget, allow the user to change the record, and then, on save, will do "something" to re-render that row. What is that "something"? Do I have to refresh the whole list? Or can I just re-render the one row? How? (that is, what's the best, default way of doing this?) 

4) How would you go, with inserting a _widget_ in each row? Now I have:

```
return domConstruct.create('div', { innerHTML:JSON.stringify(item) } );
```

Should I do this:

```
return new SomeObject( { someParameter: item.name } ).domNode );
```

And then retrieve the widget with dojo/registry?

Thanks a lot...! 

Merc.

P.S.
Basically, unless I _am_ doing something terribly wrong (1) and as soon as I work out how to re-render (3), I think I am good to go...!
